graylog2-server/src/main/java/org/graylog2/filters/ExtractorFilter.java

```diff
@@ -70,34 +83,43 @@ public boolean filter(Message msg) {
         return false;
     }
 
-    private List<Extractor> loadExtractors(final String inputId) {
+    @Subscribe
+    public void handleInputCreate(final InputCreated event) {
+        LOG.debug("Load extractors for input <{}>", event.id());
+        scheduler.schedule(() -> loadExtractors(event.id()), 0, TimeUnit.SECONDS);
+    }
+
+    @Subscribe
+    public void handleInputDelete(final InputDeleted event) {
+        LOG.debug("Removing input from extractors cache <{}>", event.id());
+        extractors.remove(event.id());
+    }
+
+    @Subscribe
+    public void handleInputUpdate(final InputUpdated event) {
```

Comment:
Maybe some check in my IntelliJ IDEA configuration or in newer versions of IDEA?
<img width="738" src="https://cloud.githubusercontent.com/assets/43951/16954067/c185e00a-4dcf-11e6-8322-81bac7e6b819.png">
<img width="691" src="https://cloud.githubusercontent.com/assets/43951/16954078/c6b40b60-4dcf-11e6-8839-eb132f28e90d.png">